doc/api/repl.markdown

```diff
@@ -106,6 +108,14 @@ the following values:
 You can use your own `eval` function if it has following signature:
 
     function eval(cmd, context, filename, callback) {
+      var result;
+      try {
+        result = vm.runInThisContext(cmd);
+      } catch (e) {
+        if (isRecoverableError(e)) {
```

Comment:
@blakeembrey Can we use the simplified form of the one actually used by REPL. May be something like

```js
function isRecoverableError(e) {
  if (e && e.name === 'SyntaxError') {
    if (e.message === 'Missing } in template expression') {
      return true;
    }
    return /^(Unexpected end of input|Unexpected token)/.test(e.message);
  }
  return false;
}
```